### PR TITLE
build(cert): Sign Windows Executable

### PR DESCRIPTION
**Pull request by louis-bompart**
https://coveord.atlassian.net/browse/CDX-186

## Proposed changes

This PR modifies the release process to ensure that the Windows executables are certified.

## Testing

- ❌ Unit Tests: not possible
- ❌  Functional Tests: not possible
- [X] Manual Tests: check https://github.com/coveo/cli/actions/runs/959562630

-----
CDX-186

### CODE REVIEW

File: .github/workflows/build-binaries.yml [hunk 1]
```diff
@@ -50,6 +51,14 @@ jobs:
           ./node_modules/@oclif/dev-cli/bin/run pack -t="${{ matrix.package-targets }}"
           mv ./dist/version ./dist/${{ matrix.os }}-version
           mv ./dist/coveo-${{ env.tag }}/coveo-${{ env.tag }}.tar.gz ./dist/coveo-${{ env.tag }}/${{ matrix.os }}-coveo-${{ env.tag }}.tar.gz
+      - name: Sign Executable
```
Comment (olamothe) [inline .github/workflows/build-binaries.yml:54]: How do you see us mitigating the certificate expiration ?

Versus having to remember it every year ? :)

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -28,6 +28,11 @@ jobs:
         run: |
           git config --global core.autocrlf false
           git config --global core.eol lf
+      - uses: actions/upload-artifact@v2
```
Comment (olamothe) [inline .github/workflows/build.yml:31]: Why is this needed ?

**Review by olamothe — APPROVED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (louis-bompart) [reply .github/workflows/build.yml:31]: artifact, good call, cleaning!

File: .github/workflows/build-binaries.yml (hunk 1 above)
Comment (louis-bompart) [reply .github/workflows/build-binaries.yml:54]: I don't really have a fool-proof solution on that end.
I know the certificate will expire in June 2024 and that it's used by Maestro and our Nugget packages too:
https://coveord.atlassian.net/wiki/spaces/CI/pages/2514747576/Maestro+-+Code+Signing+Certificate+PFX
